platform/mbed_sleep.h

```diff
@@ -60,25 +96,27 @@ __INLINE static void sleep(void)
  * @note This function will be a noop in debug mode (debug build profile when MBED_DEBUG is defined)
  * @note This function will be a noop while uVisor is in use.
  *
- * This processor is setup ready for deep sleep, and sent to sleep using __WFI(). This mode
+ * This processor is setup ready for deep sleep, and sent to sleep. This mode
  * has the same sleep features as sleep plus it powers down peripherals and clocks. All state
  * is still maintained.
  *
- * The processor can only be woken up by an external interrupt on a pin or a watchdog timer.
+ * The processor can only be woken up by RTC, an external interrupt on a pin or a watchdog timer.
  *
+ *  The wake-up time should be less than 10 ms.
+ *  
  * @note
  *  The mbed interface semihosting is disconnected as part of going to sleep, and can not be restored.
  * Flash re-programming and the USB serial port will remain active, but the mbed program will no longer be
  * able to access the LocalFileSystem
  */
+
+MBED_DEPRECATED_SINCE("mbed-os-5.6", "One entry point for an application, use sleep()")
 __INLINE static void deepsleep(void)
 {
 #if !(defined(FEATURE_UVISOR) && defined(TARGET_UVISOR_SUPPORTED))
-#ifndef MBED_DEBUG
 #if DEVICE_SLEEP
     hal_deepsleep();
```

Comment:
But it goes around our lock/unlock logic so it can cause troubles. I would say the API should stay the same (just marked as deprecated), but which exactly sleep is taken should be decided by the new logic.